tests/testerina-integration-test/src/test/resources/project-based-tests/basic-tests/src/assertions-error-messages/tests/assertions-diff-error-messages.bal

```diff
@@ -56,18 +70,45 @@ function testAssertJsonValues() {
     json bioData2 = {name:"John Doe New", age:25, address:{city:"Colombo", country:"Sri Lanka"}};
     error? err = trap test:assertEquals(bioData, bioData2);
     error result = <error>err;
-    test:assertEquals(result.message().toString(),
-    "Assertion Failed!\n \nexpected: '{\"name\":\"John Doe New\",\"age\":25,\"address\":{\"city\":\"Colombo\"," + "\"country\":\"Sri Lanka...'\nactual\t: '{\"name\":\"John Doe\",\"age\":25,\"address\":{\"city\":\"Colombo\"," + "\"country\":\"Sri Lanka\"}}'");
+    test:assertEquals(result.message().toString(), "Assertion Failed!\n \nexpected: '{\"name\":\"John Doe New\",\"age\":25,\"address\":{\"city\":\"Colombo\"," + "\"country\":\"Sri Lanka...'\nactual\t: '{\"name\":\"John Doe\",\"age\":25,\"address\":{\"city\":\"Colombo\"," + "\"country\":\"Sri Lanka\"}}'\n \nDiff\t:\n \n\n \nkey: name\nexpected value\t: John Doe New\nactual value\t: John Doe \n \n");
+}
+
+@test:Config {}
+function testAssertJsonInJson() {
+    json j1 = {name: "Anne", age: "21", marks: {maths: 100, physics: 90, status: {pass:true}}};
+    json j2 = {name: "Anne", age: 21, marks: {maths: 100, physics: 90, status: {pass:false}}};
+    error? err = trap test:assertEquals(j1, j2);
+    error result = <error>err;
+    test:assertEquals(result.message().toString(), "Assertion Failed!\n \nexpected: '{\"name\":\"Anne\",\"age\":21,\"marks\":{\"maths\":100,\"physics\":90,\"status\":{\"pass\":false...'\nactual\t: '{\"name\":\"Anne\",\"age\":\"21\",\"marks\":{\"maths\":100,\"physics\":90,\"status\":{\"pass\":tru...'\n \nDiff\t:\n \n\n \nkey: age\nexpected value\t: <int> 21\nactual value\t: <string> 21 \n\n \nkey: marks.status.pass\nexpected value\t: false\nactual value\t: true \n \n");
 }
 
 @test:Config {}
 function testAssertLongJsonValues() {
     json bioData = {name:"John Doe Old", age:25, designation: "SSE", address:{city:"Colombo", country:"Sri Lankaa"}};
     json bioData2 = {name:"John Doe New", age:25, designation: "SSE", address:{city:"Colombo", country:"Sri Lanka"}};
+    //test:assertEquals(bioData, bioData2);
```

Comment:
Is this a description comment or something that needs to be asserted? Can we remove it?